### ISSUE TITLE
PHP Fatal error:  Uncaught Error: Class 'Aws\Common\Aws' not found in inc/class-s3-uploads-wp-cli-command.php:155

### ISSUE THREAD

**Issue by cluber22** [humanmade/S3-Uploads#166]
Doesn't seem to be able to find **Aws\Common\Aws** in class-s3-uploads-wp-cli-command.php

`$iam = Aws\Common\Aws::factory( array( 'key' => $args_assoc['admin-key'], 'secret' => $args_assoc['admin-secret'] ) )->get( 'iam' );`

```
[ec2-user@ip-172-31-9-169 mysite.com]$ ./wp s3-uploads create-iam-user --admin-key=hidden --admin-secret=hidden
PHP Fatal error:  Uncaught Error: Class 'Aws\Common\Aws' not found in /var/app/current/mysite.com/wp-content/plugins/S3-Uploads/inc/class-s3-uploads-wp-cli-command.php:155
Stack trace:
#0 [internal function]: S3_Uploads_WP_CLI_Command->create_iam_user(Array, Array)
#1 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Dispatcher/CommandFactory.php(67): call_user_func(Array, Array, Array)
#2 [internal function]: WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure}(Array, Array)
#3 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Dispatcher/Subcommand.php(390): call_user_func(Object(Closure), Array, Array)
#4 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Runner.php(316): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#5 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Runner.php(323): WP_CLI\Runner->run_command(Array, Array)
#6 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Runner.php(947): WP_CLI\Runner->_run_command()
#7 phar:///var/app/current/mysite.com/wp/php/wp-cli.php(21): WP_CLI\Runne in /var/app/current/mysite.com/wp-content/plugins/S3-Uploads/inc/class-s3-uploads-wp-cli-command.php on line 155
Fatal error: Uncaught Error: Class 'Aws\Common\Aws' not found in /var/app/current/mysite.com/wp-content/plugins/S3-Uploads/inc/class-s3-uploads-wp-cli-command.php:155
Stack trace:
#0 [internal function]: S3_Uploads_WP_CLI_Command->create_iam_user(Array, Array)
#1 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Dispatcher/CommandFactory.php(67): call_user_func(Array, Array, Array)
#2 [internal function]: WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure}(Array, Array)
#3 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Dispatcher/Subcommand.php(390): call_user_func(Object(Closure), Array, Array)
#4 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Runner.php(316): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#5 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Runner.php(323): WP_CLI\Runner->run_command(Array, Array)
#6 phar:///var/app/current/mysite.com/wp/php/WP_CLI/Runner.php(947): WP_CLI\Runner->_run_command()
#7 phar:///var/app/current/mysite.com/wp/php/wp-cli.php(21): WP_CLI\Runne in /var/app/current/mysite.com/wp-content/plugins/S3-Uploads/inc/class-s3-uploads-wp-cli-command.php on line 155
```

I have managed to use the following with no problems:

`wp s3-uploads verify`
`wp s3-uploads migrate-attachments`

Notes: 
SP-Uploads Version: v2.0.0-beta3
Platform: 64bit Amazon Linux 2016.09 v2.3.3 running PHP 7.0

**Comment by cluber22** [comment]
Note that I have managed to use the following with no problems:

`wp s3-uploads verify`
`wp s3-uploads migrate-attachments`

**Comment by ajithrn** [comment]
i got the same error while am trying to create new user

command used: 
`wp s3-uploads create-iam-user --admin-key=<key> --admin-secret=<secret>`

Error Produced:
`PHP Fatal error:  Uncaught Error: Class 'Aws\Common\Aws' not found in /srv/www/s3demo.mysite.com/current/web/app/plugins/S3-Uploads/inc/class-s3-uploads-wp-cli-command.php:155

Stack trace:
#0 [internal function]: S3_Uploads_WP_CLI_Command->create_iam_user(Array, Array)
#1 phar:///usr/bin/wp/php/WP_CLI/Dispatcher/CommandFactory.php(67): call_user_func(Array, Array, Array)
#2 [internal function]: WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure}(Array, Array)
#3 phar:///usr/bin/wp/php/WP_CLI/Dispatcher/Subcommand.php(372): call_user_func(Object(Closure), Array, Array)
#4 phar:///usr/bin/wp/php/WP_CLI/Runner.php(312): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#5 phar:///usr/bin/wp/php/WP_CLI/Runner.php(319): WP_CLI\Runner->run_command(Array, Array)
#6 phar:///usr/bin/wp/php/WP_CLI/Runner.php(906): WP_CLI\Runner->_run_command()
#7 phar:///usr/bin/wp/php/wp-cli.php(21): WP_CLI\Runner->start()
#8 phar:///usr/bin/wp/php/boot-phar.php(5): include('phar:///usr/bin...')
#9 /usr/bin/wp(4): include('pha in /srv/www/s3demo.mysite.com/current/web/app/plugins/S3-Uploads/inc/class-s3-uploads-wp-cli-command.php on line 155`

Server based on [https://roots.io/trellis/](https://roots.io/trellis/)

**Comment by rmccue** [comment]
I think the command may need updating @joehoyle?

**Comment by joehoyle** [comment]
@cluber22 @ajithrn this should be fixed in the latest 2.0 beta
